### DISCUSSION TITLE
Possible to create editable Multi-Row-Per-Row Datagrid, also with dropdown field in one cell of the grid?

### DISCUSSION THREAD

**Discussion by bswylie** [AvaloniaUI/Avalonia#14834] [Q&A] [answered]
Hi everyone - I'm new to Avalonia UI and hopeful to use it after trying out the samples and tuts. I'm hoping to be able to achieve something like the below screenshot, where each row of the datagrid is effectively a "compound row", as pushing this into too many columns just won't be a great interface. Additionally, the date cell would of course be a date picker, and I'd like one of the columns to be a searchable dropdown list. 

Are these things possible in Avalonia UI's Datagrid? Are there other ways to achieve this? (I've considered Stackpanel perhaps but my gut tells me the datagrid is the better / "correct" way to go?)

Any guidance would be greatly appreciated, thank you!

<img width="1062" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/15008951/2d7b690c-2a1d-4c03-9529-7c46e70f5ac8">

**Comment by timunie** [comment, marked as answer]
consider using `DataGridTemplateColumns`. You can put inside whatever you need, also two "rows"

**Reply by bswylie** [reply]
Many thanks for the quick response, Tim! I noticed that reference in the API documentation but it didn’t sound like what I was looking for so I ignored it. Will look into it!

**Reply by timunie** [reply]
We don't have an Excel-like join cells option, so that's the best option I see here.

**Comment by thevortexcloud** [comment]
Avalonia's Datagrid also supports grouping. But it may not do what you want.

**Reply by bswylie** [reply]
Thank you! I will also look into understanding that!
